freshmaker/pyxis.py

```diff
@@ -0,0 +1,246 @@
+                        old_ver = bundle_per_channel[channel]['version']
+                        if version.Version(new_ver) > version.Version(old_ver):
+                            bundle_per_channel[channel] = bundle
+                    else:
+                        # Check if the right format of version is used
+                        version.Version(bundle.get('version'))
+                        bundle_per_channel[channel] = bundle
+                # Check if the right format of version is used
+                except version.InvalidVersion as e:
+                    path = bundle.get('bundle_path', 'Unknown bundle path')
+                    log.warning("Other format than SemVer is used in "
+                                "bundle: %s", path)
+                    log.warning(repr(e))
+            ret_bundles.extend(bundle_per_channel.values())
+
+        return ret_bundles
+
+    def get_digests_by_nvrs(self, nvrs, params=None):
+        """
+        Get images' digests(manifest_list_digest field) by their nvrs
+
+        :param list nvrs: list of nvrs of ContainerImages to query Pyxis
+        :param dict params: parameters to add to GET request
+        :return: set of digests
+        :rtype: set
+        """
+        if params is None:
+            request_params = {}
+        else:
+            request_params = copy.deepcopy(params)
+        request_params['include'] = ','.join([
+            'data.brew', 'data.repositories'])
+
+        # get all manifest_list_digest of ContainerImages we got from Pyxis
+        digests = set()
+        for nvr in nvrs:
+            for image in self._pagination(f'images/nvr/{nvr}', request_params):
+                for repo in image.get('repositories'):
+                    if repo['published'] and 'manifest_list_digest' in repo:
+                        digests.add(repo['manifest_list_digest'])
+        return digests
+
+    def filter_bundles_by_related_image_digests(self, original_digests,
+                                                bundles):
+        """
+        Filter out bundles that don't have same digests in 'related_image'
+        as manifest list digests in 'original_digests'
+
+        :param set original_digests: digests of the original operator/operand
+            images to filter bundles by them
+        :param list bundles: bundles to filter
+        :return: filtered list of bundles
+        :rtype: list
+        """
+        ret_bundles = []
+        # If bundle doesn't have any of digests of images, don't add it to return list
+        for bundle in bundles:
+            for image in bundle.get('related_images', []):
+                # If same digest within images' digests is found, it will be in return list
+                if image.get('digest') in original_digests:
+                    ret_bundles.append(bundle)
+                    break
+
+        return ret_bundles
+
+    def get_images_by_digests(self, digests, params=None):
```

Comment:
It seems `params` isn't ever used. Could you please remove it? It's best to not add unused code. We can always add it when needed.